packages/wrangler/src/dev.tsx

```diff
@@ -811,15 +818,52 @@ export async function startDev(args: StartDevOptions) {
 		}
 
 		if (config.configPath && !args.experimentalDevEnv) {
-			watcher = watch(config.configPath, {
+			configFileWatcher = watch(config.configPath, {
 				persistent: true,
 			}).on("change", async (_event) => {
-				// TODO: Do we need to handle different `_event` types differently?
-				//       e.g. what if the file is deleted, or added?
-				config = readConfig(configPath, args);
-				if (config.configPath) {
+				try {
+					// TODO: Do we need to handle different `_event` types differently?
+					// e.g. what if the file is deleted, or added?
+					config = readConfig(configPath, args);
+					if (!config.configPath) {
+						return;
+					}
+
 					logger.log(`${path.basename(config.configPath)} changed...`);
+
+					/*
+					 * Handle experimental assets watching on config file changes
+					 *
+					 * 1. if experimental assets was specified via CLI args, config file
```

Comment:
I wonder if for this use case we shouldn't show a warning or log message reminding users that they are currently running `wrangler dev` with the `-x--assets`/`--experimental-assets` flag, and therefore config file changes related to `experimental_assets` will not be reflected in the current dev process 🤔

Reply:
as per our async convo, I will put smth in place here